File: api/scala/api_to_json.scala
import scala.io.Source
import java.io.PrintWriter
import scala.util.{Try, Success, Failure}
import scala.concurrent.ExecutionContext.Implicits.global
import scala.concurrent.Future
import scala.concurrent.Await
import scala.concurrent.duration._

import scala.language.postfixOps

import scala.concurrent.duration._

object Main extends App {
  val url = "https://randomuser.me/api/"
  val filePath = "data.json"

  val futureResponse: Future[String] = Future {
    Source.fromURL(url).mkString
  }

  val result = Try(Await.result(futureResponse, 5 seconds))

  result match {
    case Success(responseBody) =>
      val writer = new PrintWriter(filePath)
      writer.write(responseBody)
      writer.close()
      println("File saved successfully.")

    case Failure(exception) =>
      println(s"An error occurred: ${exception.getMessage}")
  }
}

File: api/python/api_to_json.py
import requests

def main():
    url = "https://randomuser.me/api/"
    try:
        response = requests.get(url)
        response.raise_for_status()
        data = response.text
        
        with open("data.json", "w") as file:
            file.write(data)
            
    except requests.exceptions.RequestException as e:
        raise SystemExit(e)

if __name__ == "__main__":
    main()
